src/modules/pinservice.ts

```diff
@@ -0,0 +1,44 @@
+/* eslint-disable import/prefer-default-export */
+import fetch from "cross-fetch";
+
+export const pin = async (cid: string) => {
+  try {
```

Comment:
If you add a non exposed "query api" function that covers headers & error catching for all requests you can save at least 10 lines of code ;)

Reply:
changed